### PR TITLE
[docs] Improve Upgrade plan docs

### PR DESCRIPTION
**Pull request by oliviertassinari**
An attempt to improve #5683. Here is what I found as opportunities in https://mui.com/x/introduction/licensing/#upgrading-from-community

1. The TOCs, there are harder to scan, using "Upgrading" would be faster to search and scan. We use two different h2, for the same information, It feels more cluttered. 

Before

Screenshot 2022-09-03 at 17 11 20

After

Screenshot 2022-09-03 at 17 09 26

2. "Superset", this part seems super important, I'm giving it more emphasis and also make it clear that this is a rule we aim to always follow.
3. We break the superset rule in one case, this also feels important to document.
4. We are in the docs for the license, but the docs feels like it's about the data grid. I have turned the phrase so it takes the data grid as an example
5. "MUI X (Pro or Premium)" -> "MUI X Pro or Premium" to lean on the side of MUI X being an open-source product on its own.
6. "Upgrading from Community" -> "Upgrading from Community to Pro" to be clearer.
7. I had to read the "You can use your Pro license as a credit when purchasing MUI X Premium. We'll provide you with a discount based on the remaining time that your current license is valid. Please contact us at [sales@mui.com](mailto:sales@mui.com?subject=My%20upgrade%20discount%20to%20Premium) to upgrade." info twice to understand what we were talking about. I have tried to rephrase it.

https://deploy-preview-6018--material-ui-x.netlify.app/x/introduction/licensing/#upgrading

### CODE REVIEW

File: docs/data/introduction/licensing/licensing.md [hunk 1]
```diff
@@ -56,39 +56,41 @@ Premium npm package:
 The features exclusive to the Premium version are marked with the <span class="plan-premium"></span> icon throughout the documentation.
 :::
 
-## Upgrading from Community
+## Upgrading
 
-When you purchase a license for MUI X (Pro or Premium), it will unlock new [packages](#plans) you can use on your project.
+The npm packages of of any given plan is a **superset** of the npm packages of the plan below.
```
Comment (joserodolfofreitas) [inline docs/data/introduction/licensing/licensing.md:61]:
```suggestion
The npm package of any given plan is a **superset** of the package on the plan below.
```

File: docs/data/introduction/licensing/licensing.md [hunk 2]
```diff
@@ -56,39 +56,41 @@ Premium npm package:
...
+So to upgrade, replace the [npm packages](#plans) you installed and imported with the ones of the plan above.
```
Comment (joserodolfofreitas) [inline docs/data/introduction/licensing/licensing.md:62]:
```suggestion
So to upgrade, replace the [npm packages](#plans) and the components' imports with the ones from the target plan.
```

File: docs/data/introduction/licensing/licensing.md [hunk 3]
```diff
@@ -56,39 +56,41 @@ Premium npm package:
...
 
-If you're using the data grid, the commercial npm packages (`@mui/x-data-grid-pro` and `@mui/x-data-grid-premium`) are a superset of the community.
+For example with the Data Grid:
```
Comment (joserodolfofreitas) [inline docs/data/introduction/licensing/licensing.md:64]:
```suggestion
For example, when you want to upgrade the Data Grid:
```

File: docs/data/introduction/licensing/licensing.md [hunk 4]
```diff
@@ -56,39 +56,41 @@ Premium npm package:
...
+  -import { DataGridPro } from '@mui/x-data-grid-pro';
+  +import { DataGridPremium } from '@mui/x-data-grid-premium';
+  ```
 
--import { DataGridPro } from '@mui/x-data-grid-pro';
-+import { DataGridPremium } from '@mui/x-data-grid-premium';
-```
+  :::info
+  If you are looking for upgrading from Pro to Premium, please contact us at [sales@mui.com](mailto:sales@mui.com?subject=My%20upgrade%20discount%20to%20Premium).
+  We'll provide you with a discount to recoup the remaining time of your Pro license.
```
Comment (joserodolfofreitas) [inline docs/data/introduction/licensing/licensing.md:90]:
```suggestion
  We'll provide you with a discount based on the remaining time of your current license term.
```

File: docs/data/introduction/licensing/licensing.md [hunk 5]
```diff
@@ -56,39 +56,41 @@ Premium npm package:
...
-For more details on how to install the packages, please check out our [package installation guide](/x/introduction/installation/).
+  :::warning
+  ⚠️ However, there is an exception to the superset rule. The default value of the `pagination` prop changes, [see the docs of the pagination](https://mui.com/x/react-data-grid/pagination/).
+  :::
 
-## Upgrading from Pro to Premium
+- **Upgrading from Pro to Premium.**
 
-When upgrading from Pro, you can upgrade the dependency on `@mui/x-data-grid-pro` to `@mui/x-data-grid-premium` and replace all your data grid imports.
+  `@mui/x-data-grid-premium` is a superset of `@mui/x-data-grid-pro`, so you can upgrade from the Pro to the Pro Premium like this:.
```
Comment (joserodolfofreitas) [inline docs/data/introduction/licensing/licensing.md:81]:
```suggestion
  `@mui/x-data-grid-premium` is a superset of `@mui/x-data-grid-pro`, so you can upgrade from Pro to Premium like this:
```

**Review by joserodolfofreitas — COMMENTED** [review]
Nice iteration. It's indeed less cluttered.
Leaving some minor suggestions to further iterate on it.

**Comment by oliviertassinari** [comment]
@joserodolfofreitas Thanks for the polish, I missed them!

**Pull request merged**